src/QueryFilterHelpers.ts

```diff
@@ -0,0 +1,604 @@
+ * Gets a list of codes in a CQL concept reference by name. These are returned as
+ * FHIR Codings.
+ *
+ * @param name Name of the concept.
+ * @param library The library elm the concept should be found in.
+ * @returns A list of codings in the concept
+ */
+function getCodesInConcept(name: string, library: ELM): R4.ICoding[] {
+  const concept = library.library.concepts?.def.find(concept => concept.name === name);
+  if (concept) {
+    const codes: R4.ICoding[] = [];
+    concept.code.map(codeRef => {
+      const code = library.library.codes?.def.find(code => code.name == codeRef.name);
+      if (code) {
+        codes.push({
+          code: code?.id,
+          system: library.library.codeSystems.def.find((systemRef: any) => code?.codeSystem.name === systemRef.name).id
+        });
+      }
+    });
+    return codes;
+  }
+  return [];
+}
+
+/**
+ * Parses an ELM equivalent expression into a filter. This currently only handles checks against literal values.
+ *
+ * @param equal The equal expression to be parsed.
+ * @returns Filter representing the equal filter.
+ */
+function interpretEqual(equal: ELMEqual): EqualsFilter | UnknownFilter {
+  let propRef: ELMProperty | null = null;
+  if (equal.operand[0].type == 'FunctionRef') {
+    propRef = interpretFunctionRef(equal.operand[0] as ELMFunctionRef);
+  } else if (equal.operand[0].type == 'Property') {
+    propRef = equal.operand[0] as ELMProperty;
+  }
+
+  let literal: ELMLiteral | null = null;
+  if (equal.operand[1].type == 'Literal') {
+    literal = equal.operand[1] as ELMLiteral;
+  }
+
+  if (propRef?.scope && literal?.value) {
+    return {
+      type: 'equals',
+      alias: propRef.scope,
+      value: literal.value,
+      attribute: propRef.path,
+      localId: equal.localId
+    };
+  } else {
+    console.log('could not find attribute or literal for Equal');
+  }
+  return { type: 'unknown' };
+}
+
+/**
+ * Parses an `IncludedIn` clause to a simpler filter representation. Currently handles comparisons to the
+ * "Measurement Period" parameter.
+ *
+ * @param includedIn IncludedId expression to be parsed.
+ * @param library The library the expression resides in.
+ * @param parameters The original parameters used for calculation.
+ * @returns Filter representation of the IncludedIn clause.
+ */
+function interpretIncludedIn(includedIn: ELMIncludedIn, library: ELM, parameters: any): DuringFilter | UnknownFilter {
+  let propRef: ELMProperty | null = null;
+  if (includedIn.operand[0].type == 'FunctionRef') {
+    propRef = interpretFunctionRef(includedIn.operand[0] as ELMFunctionRef);
+  } else if (includedIn.operand[0].type == 'Property') {
+    propRef = includedIn.operand[0] as ELMProperty;
+  }
+
+  if (propRef == null) {
+    console.warn(
+      `could not resolve property ref for IncludedIn:${includedIn.localId}. first operand is a ${includedIn.operand[0].type}`
+    );
+    return { type: 'unknown' };
+  }
+
+  if (includedIn.operand[1].type == 'ParameterRef') {
```

Comment:
So we're only supporting Measurement Period parameter in this interpret for now, but is there ever a case where we'd get in here, and the parameter type wouldn't be a `cql.Interval`? Basically, I'm asking if the right side of an `IncludedIn` expression is always going to be an interval in CQL.

If not, we'd probably want to fail gracefully here, since otherwise it might say something like ".start() is not a function" which isn't as useful of a message as "Could not parse parameter of type ___ or something"

Reply:
This is a good point. Parameters can be of any type valid to be used in CQL. Could even be a FHIR resource. I'll add a check to make sure it's an interval.

Reply:
Added a check on the parameter type and created tests for that situation in 9928301.